### DISCUSSION TITLE
Control of stepper motors is broken

### DISCUSSION THREAD

**Discussion by canis-ferrin** [arendst/Tasmota#21758] [Q&A]
Issue:
The control of stepper motors is broken in v14.0.0.2 Rodney. 
If you type in ShutterMotorDelay1 2.5 you get an error message.
I tried v12.0.2 Paul, where it is working.
But also, it is not possiblle to control power3 manually, which is nessesary to lock thge motor in up position.
To repuce, try as written in https://github.com/arendst/Tasmota/discussions/11731

**Comment by canis-ferrin** [comment]
Same issue in 14.1.0.3

**Comment by canis-ferrin** [comment]
Another mistake in v12.0.2 Paul: When the device is powered on the motor moves downwards and more as the adjusted distance

**Comment by stefanbode** [comment]
Please can you send more logs. I operate two doors with stepper on 14.1.2 and do not see any problems. Anyhow this is ESP32. On ESP8266 this can be different. Please open an ISSUE and share log files and configuration that help us to find the root cause and fix it. Happy. to help

![image](https://github.com/arendst/Tasmota/assets/16225290/48cb5b40-b32c-4b46-89b9-73d76edd32b1)

**Comment by canis-ferrin** [comment]
I started from beginning: Tasmota 14.1.0.3 unpacked, with Visual Studio Code my_user_config.h adjusted, uploaded. I followed https://github.com/arendst/Tasmota/discussions/11731...
19:08:38.262 RSL: SENSOR = {"Time":"2024-07-11T19:08:38","COUNTER":{"C1":0}}
19:08:46.655 CMD: ShutterFrequency 2500
19:08:46.661 RSL: RESULT = {"ShutterFrequency":5000}
19:08:58.762 CMD: ShutterMotorDelay1 2.5
19:08:58.768 RSL: RESULT = {"Command":"Error"}
A4988 activated, result the same:
Command ShutterMotorDelay1 2.5 delivers
19:19:18.416 CMD: ShutterMotorDelay1 2.5
19:19:18.424 MQT: Garten/Grillwinde2/stat/RESULT = {"Command":"Error"}

**Comment by canis-ferrin** [comment]
Thanks for advise. I orderd a ESP32 and will try it next week, to see, if there are differences to the ESP8266.

**Comment by canis-ferrin** [comment]
I received an ESP32 with an expansion board yesterday evening and installed Tasmota today.
It is definitely a bug in the newer ESP8266 Tasmota versions,
with ESP32 the shutter and blinds control works with a stepper motor. (Like with the old version 12),
now the slider also moves.
However, I didn't manage to get the motor locked with the "ShutterStopRelay1 3" command, the console outputs
/stat/RESULT = {"ShutterStopRelay1":0} or /stat/RESULT = {"ShutterStopRelay1":1}, but the motor is not blocked.
The behavior is the same for versions 14.1.0 and 14.1.0.3.

**Reply by stefanbode** [reply]
Yes, this stop relay initially was created for garage doors that had a stop relay separat. A look into the code looks promising it do also the trick with steppers. Thanks for the feedback. Looks like there is some tailoring required. 

The relay4 issue is interesting. Never seen this before. Is relay 4 defined in an interlock? A log file would really help to get an idea because this is very strange.

Regarding the stoprelay: just let me confirm the use case. You want the stepper keep under power to prevent any accidental movement because the is torque load on the wheel?

**Comment by canis-ferrin** [comment]
Maybe, it's for intrest: ShutterStopRelay1 is not working.
When POWER4 (or the relay with the highest number) is toggled from on to off during moving at full speed, the motor stops abruptly and is locked.
But this is not working during ramp movement.
My project isset to suspended.

**Comment by canis-ferrin** [comment]
Problem solved. 
After thinking about this matter a night long on the road I defined an additional relais and exchanged it with the ENA relais.
Now motor is locked when Toggle 3 to ON.
To operate the motor Toggle 3 has to be ON to enable the motor, after operation it can be switched OFF to unlock the motor.
